test/infrastructure/container/interface.go

```diff
@@ -128,3 +133,15 @@ type Container struct {
 	// Status is the status of the container
 	Status string
 }
+
+// RuntimeFromContext is used to extract the container runtime client from a
```

Comment:
I like the consistency - I'll update the names.

As far as making it a function (rather than a method on the actual Runtime instance) that gets tricky. The context takes a reference to the instance, but I couldn't figure out how to make it happy with a generic `Runtime` interface. For now I'm keeping it on the instances themselves, but if you have any ideas how to make that work... I guess that's one area my Go-fu is falling short. ;)